libass/ass_font.c

```diff
@@ -427,20 +427,23 @@ static int add_face(ASS_FontSelector *fontsel, ASS_Font *font, uint32_t ch)
     FT_Face face;
     int ret = -1;
 
+    ass_fontselect_lock(fontsel);
+
     if (font->n_faces == ASS_FONT_MAX_FACES)
-        return -1;
+        goto fail;
 
     path = ass_font_select(fontsel, font, &index,
             &postscript_name, &uid, &stream, ch);
 
     if (!path)
-        return -1;
+        goto fail;
 
     for (i = 0; i < font->n_faces; i++) {
         if (font->faces_uid[i] == uid) {
             ass_msg(font->library, MSGL_INFO,
                     "Got a font face that already is available! Skipping.");
-            return i;
+            ret = i;
+            goto fail;
```

Comment:
nit: this label is now used for a non-failing condition here; may be better to rename

Reply:
Sure, done.